toolkit/tools/pkg/imagecustomizerlib/cosicommon.go

```diff
@@ -188,5 +202,41 @@ func populateMetadata(data ImageBuildData) error {
 		return fmt.Errorf("failed to calculate sha384 of %s:\n%w", data.Source, err)
 	}
 	data.Metadata.Image.Sha384 = sha384
+
+	err = populateVerityMetadata(data.VeritySource, data.Metadata.Verity)
+	if err != nil {
+		return fmt.Errorf("failed to populate verity metadata: %w", err)
+	}
+	return nil
+}
+
+
+func populateVerityMetadata(source *string, verity *Verity) error {
+	if source == nil && verity == nil {
+		return nil
+	}
+
+	if source == nil || verity == nil {
+		return fmt.Errorf("source and verity file must be both defined or both undefined")
+	}
+
+	verityFile := &verity.Image
+
+	verityStat, err := os.Stat(*source)
+	if err != nil {
+		return fmt.Errorf("failed to stat verity source: %w", err)
+	}
+
+	verityFile.CompressedSize = uint64(verityStat.Size())
+
+	veritySha384, err := sha384sum(*source)
+	if err != nil {
+		return fmt.Errorf("failed to calculate sha384 of verity source: %w", err)
+	}
+
+	verityFile.Sha384 = veritySha384
+
+	verityFile.UncompressedSize = uint64(verityStat.Size())
```

Comment:
Isn't this the same as the compressed size?

Reply:
+1. You need to pull the `UncompressedSize` from the `[]outputPartitionMetadata` list.